src/SignalR/clients/ts/signalr/tests/HubConnectionBuilder.test.ts

```diff
@@ -111,98 +134,160 @@ describe("HubConnectionBuilder", () => {
+            const capturingConsole = new CapturingConsole();
+            (logger as ConsoleLogger).outputConsole = capturingConsole;
+
+            for (let level = LogLevel.Trace; level < LogLevel.None; level++) {
+                const message = `Message at LogLevel.${LogLevel[level]}`;
+                const expectedMessage = `${LogLevel[level]}: Message at LogLevel.${LogLevel[level]}`;
+                logger.log(level, message);
+
+                if (level >= minLevel) {
+                    expect(capturingConsole.messages).toContain(expectedMessage);
+                } else {
+                    expect(capturingConsole.messages).not.toContain(expectedMessage);
+                }
+            }
         }
 
-        expect(loggedMessages).toBeGreaterThan(0);
-    });
+        eachMissingValue((val, name) => {
+            it(`throws if logger is ${name}`, () => {
+                const builder = new HubConnectionBuilder();
+                expect(() => builder.configureLogging(val!)).toThrow("The 'logging' argument is required.");
+            });
+        });
 
-    it("uses logger for both HttpConnection and HubConnection", async () => {
-        const pollSent = new PromiseSource<HttpRequest>();
-        const pollCompleted = new PromiseSource<HttpResponse>();
-        const testClient = createTestClient(pollSent, pollCompleted.promise)
-            .on("POST", "http://example.com?id=abc123", (req) => {
-                // Respond from the poll with the handshake response
-                pollCompleted.resolve(new HttpResponse(204, "No Content", "{}"));
-                return new HttpResponse(202);
+        [
+            LogLevel.None,
+            LogLevel.Critical,
+            LogLevel.Error,
+            LogLevel.Warning,
+            LogLevel.Information,
+            LogLevel.Debug,
+            LogLevel.Trace,
+        ].forEach((minLevel) => {
+            const levelName = LogLevel[minLevel];
+            it(`accepts LogLevel.${levelName}`, async () => {
+                const builder = new HubConnectionBuilder()
+                    .configureLogging(minLevel);
+
+                expect(builder.logger).toBeDefined();
+                expect(builder.logger).toBeInstanceOf(ConsoleLogger);
+
+                testLogLevels(builder.logger!, minLevel);
             });
-        const logger = new CaptureLogger();
-        const connection = createConnectionBuilder(logger)
-            .withUrl("http://example.com", {
-                ...commonHttpOptions,
-                httpClient: testClient,
-            })
-            .build();
-
-        try {
-            await connection.start();
-        } catch {
-            // Ignore failures
-        }
+        });
 
-        // A HubConnection message
-        expect(logger.messages).toContain("Starting HubConnection.");
+        const levelNames = Object.keys(LogLevelNameMapping) as Array<keyof typeof LogLevelNameMapping>;
```

Comment:
Maybe use a different mapping object for the tests, so test will fail if we make a breaking change without realizing.

Reply:
That's reasonable. I was thinking exactly the opposite (to save having to update the test) but I think you're right, it's better to know.